cpp/cg_method/_cg_method.hpp

```diff
@@ -0,0 +1,47 @@
+#ifndef __CG_METHOD_HPP__
+#define __CG_METHOD_HPP__
+
+#include <vector>
+#include "../matrix/_matrix.hpp"
```

Comment:
It's usually a bad idea to use `#include "..."`.  It's easy to have conflict names of include files.

Reply:
cpp
│   _cg_method.cpp
│   _cg_method.hpp
│   _matrix.cpp
│   _matrix.hpp
|    _pybind.cpp

Reply:
No.  I meant to use angle brackets as much as possible: https://stackoverflow.com/questions/21593/what-is-the-difference-between-include-filename-and-include-filename